src/config.rs

```diff
@@ -142,6 +142,8 @@ pub struct Workers {
 /// Represents the middleware configuration structure.
 #[derive(Debug, Clone, Deserialize, Serialize)]
 pub struct Middlewares {
+    /// Middleware that enable compression for the response.
+    pub compression: Option<EnableMiddleware>,
```

Comment:
looks like the correct location. instead of `EnableMiddleware` maybe you want `CompressionMiddleware` ?

Reply:
Isn't `EnableMiddleware` the generic `boolean` type?
Doesn't matter to me, whichever you think is best.

Reply:
yes, my bad let me rephrase

* if you have 0 configuration parameters - yes use EnableMiddleware which only controls switching on/off the middleware
* if you have additional specific configuration for the middleware, use your own type (and add a similarly named boolean for enable)

I was almost sure a compression middleware has some custom configuration but i may be mistaken -- its your call to decide

Reply:
Technically the compression layer can be dissected into supported formats but I don't think we care that much and just enabling it as a whole is simple and good.